tests/unit/src/commands/initialize/initAuthorization.ts

```diff
@@ -0,0 +1,184 @@
+			deleteAuthorization: SinonStub = travisDeleteAuthorizationStub;
+			fetchRepository: SinonStub = fetchRepositoryStub;
+		};
+
+		GitHub = class {
+			constructor() {}
+			owner = 'dojo';
+			name = 'grunt-dojo2-extras';
+			api: any = {
+				authenticate: authenticateStub,
+				misc: {
+					getRateLimit: getRateLimitStub
+				}
+			};
+			toString: SinonStub = toStringStub;
+			createAuthorization = repoCreateAuthorizationStub;
+			deleteAuthorization = repoDeleteAuthorizationStub;
+		};
+
+		TravisSpy = spy(Travis);
+		GitHubSpy = spy(GitHub);
+	},
+
+	after() {
+		cleanupModuleMocks();
+	},
+
+	beforeEach() {
+		travisCreateAuthorizationStub.returns(Promise.resolve());
+		travisDeleteAuthorizationStub.returns(Promise.resolve());
+		fetchRepositoryStub.returns(Promise.resolve({
+			listEnvironmentVariables: listEnvironmentVariablesStub.returns(Promise.resolve([])),
+			setEnvironmentVariables: setEnvironmentVariablesStub.returns(Promise.resolve())
+		}));
+		getRateLimitStub.returns(Promise.resolve());
+		repoCreateAuthorizationStub.returns(Promise.resolve({
+			token: 'token'
+		}));
+		repoDeleteAuthorizationStub.returns(Promise.resolve());
+
+		initAuthorization = loadModule('src/commands/initialize/initAuthorization', {
+			'../../util/Travis': { default: TravisSpy },
+			'../../util/GitHub': { default: GitHubSpy },
+			'@dojo/shim/array': {
+				find: findStub
+			},
+			'../../util/environment': {
+				env: {
+					githubAuth: githubAuthStub
+				}
+			}
+		});
+	},
+
+	afterEach() {
+		TravisSpy.reset();
+		GitHubSpy.reset();
+		isAuthorizedStub.reset();
+		travisCreateAuthorizationStub.reset();
+		travisDeleteAuthorizationStub.reset();
+		listEnvironmentVariablesStub = stub();
+		setEnvironmentVariablesStub = stub();
+		fetchRepositoryStub.reset();
+		authenticateStub.reset();
+		getRateLimitStub.reset();
+		toStringStub.reset();
+		repoCreateAuthorizationStub.reset();
+		repoDeleteAuthorizationStub.reset();
+		findStub.reset();
+		githubAuthStub.reset();
+	},
+
+	'initAuthorization': (() => {
+		return {
+			async 'explicit Travis instance'() {
+				const travis = new Travis();
+
+				await assertInitAuthorization(travis);
+			},
+
+			async 'Travis is not authorized'() {
+				isAuthorizedStub.returns(false);
+
+				await assertInitAuthorization();
+
+				assert.isTrue(travisCreateAuthorizationStub.calledOnce);
+			},
+
+			async 'fetch repo and environment, create authorization'() {
+				await assertInitAuthorization();
+
+				assert.isTrue(fetchRepositoryStub.calledOnce);
+				assert.isTrue(listEnvironmentVariablesStub.calledOnce);
+				assert.isTrue(findStub.calledOnce);
+				assert.isTrue(repoCreateAuthorizationStub.calledOnce);
+				assert.isTrue(setEnvironmentVariablesStub.calledOnce);
+			},
+
+			async 'eventually throws'() {
+				fetchRepositoryStub.returns(Promise.reject(new Error('error')));
+
+				try {
+					await assertInitAuthorization();
+
+					assert.fail();
```

Comment:
This will be caught by the catch and the test will pass anyway. You'll need to return the promise to make this work:

```
const promise = await assertInitAuthorization();

return promise.then(assert.fail, (e) => { assert.strictEqual(e.message, 'error') });
```